### PR TITLE
Sphinx Doc-String Fixes

### LINKED ISSUE #89: Code documentation: format docstrings to Google standard and avoid Sphinx warnings
As an IMPROVER developer I want docstrings in Google format and parsable by Sphinx without warnings so that I can be confident that I have well-presented code documentation. I want to capture warnings as part of the testing process of new code.

Related issues: #88

The docstring style should be in line with the [Code Style Guide](https://github.com/metoppv/improver/wiki/Code-Style-Guide#plugins).

Acceptance criteria:
* Fix any remaining inconsistencies with the code style guide above so that no warnings are produced by "improver tests doc" on master.
 * Alter `improver tests` such that any warnings fail the test
 * Check that the resulting documentation renders correctly.

### PR DESCRIPTION
**Pull request by bayliffe**
#89 Issue.

All doc-strings updated to be of the same format, including the use of tuple statements when returning multiple arguments.

This is a horribly large PR as there were many varying styles in our doc-strings. This PR does not fix all discrepencies (for example in what we call various types (e.g. Iris cube, iris.cube.Cube, Float, float, int, integer, etc.)), but it does attempt to make the formatting consistent. The consistent format is that which works with Sphinx for the generation of documentation.

I will write up a more detailed set of notes on how we should format doc-strings (with examples) now I have completed this work.

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

File: lib/improver/ensemble_calibration/ensemble_calibration_utilities.py [hunk 1]
```diff
@@ -44,23 +44,21 @@ def convert_cube_data_to_2d(
...
-        This will result in a 2d array of format [:, coord].
-
-    Returns
-    -------
-    forecast_data : Numpy array
-        Reshaped 2d array.
+    Args:
+        forecast (Iris cube):
+            N-dimensional cube to be reshaped.
+        coord (String):
```
Comment (carolinejjones) [inline lib/improver/ensemble_calibration/ensemble_calibration_utilities.py]: String should be string

This is repeated throughout this module. I shall not add a comment for every one.

File: lib/improver/ensemble_calibration/ensemble_calibration_utilities.py [hunk 2]
```diff
@@ -44,23 +44,21 @@ def convert_cube_data_to_2d(
...
+            The data will be flattened along this coordinate.
+        transpose (Logical):
```
Comment (carolinejjones) [inline lib/improver/ensemble_calibration/ensemble_calibration_utilities.py]: boolean not Logical

File: lib/improver/ensemble_calibration/ensemble_calibration_utilities.py [hunk 3]
```diff
@@ -82,14 +80,13 @@ def ensure_dimension_is_the_zeroth_dimension(cube, coord):
...
-    Parameters
-    ----------
-    cube : Iris cube
-        Cube where the requirement for the required dimension to be the first
-        dimension will be enforced.
-    coord : String
-        Name of the coordinate that is to be made the first dimension
-        coordinate in the cube.
+    Args:
+        cube (Iris cube):
```
Comment (carolinejjones) [inline lib/improver/ensemble_calibration/ensemble_calibration_utilities.py]: iris.cube.Cube to be consistent with other modules.

File: lib/improver/ensemble_copula_coupling/ensemble_copula_coupling_utilities.py [hunk 4]
```diff
@@ -52,21 +52,19 @@ def concatenate_2d_array_with_2d_array_endpoints(
...
-        as the upper endpoint.
-    Returns
-    -------
-    array_2d : Numpy array
-        2d array of values after padding with the low_endpoint and
-        high_endpoint.
+    Args:
+        array_2d (Numpy array):
+            2d array of values
+        low_endpoint (Number):
```
Comment (carolinejjones) [inline lib/improver/ensemble_copula_coupling/ensemble_copula_coupling_utilities.py]: float or int?

File: lib/improver/ensemble_copula_coupling/ensemble_copula_coupling_utilities.py [hunk 5]
```diff
@@ -81,18 +79,17 @@ def choose_set_of_percentiles(no_of_percentiles, sampling="quantile"):
...
-    sampling : String
-        Type of sampling of the distribution to produce a set of
-        percentiles e.g. quantile or random.
-        Accepted options for sampling are:
-        Quantile: A regular set of equally-spaced percentiles aimed
-                  at dividing a Cumulative Distribution Function into
-                  blocks of equal probability.
-        Random: A random set of ordered percentiles.
+    Args:
+        no_of_percentiles (Int):
```
Comment (carolinejjones) [inline lib/improver/ensemble_copula_coupling/ensemble_copula_coupling_utilities.py]: int

File: lib/improver/spotdata/extrema.py [hunk 6]
```diff
@@ -248,18 +245,16 @@ def get_datetime_limits(time_coord, start_hour):
...
-        The hour on a 24hr clock at which to set the returned times.
+        start_hour (int):
+            The hour on a 24hr clock at which to set the returned times.
 
     Returns:
-    --------
-    start_time, end_time : datetime.datetime object
-        First day and last day on a time coordinate, with the time on each day
-        set to the hour given by start hour
+        start_time, end_time (datetime.datetime object):
```
Comment (carolinejjones) [inline lib/improver/spotdata/extrema.py]: (tuple)

File: lib/improver/spotdata/neighbour_finding.py [hunk 7]
```diff
@@ -63,19 +63,18 @@ def __init__(self, method='fast_nearest_neighbour',
...
-            Name of the method of neighbour finding to be used.
+            method : string
+                Name of the method of neighbour finding to be used.
 
-        vertical_bias : string/None
-            Sets the preferred vertical displacement bias of the grid point
-            relative to the site; above/below/None. If this criteria cannot be
-            met (e.g. bias below, but all grid points above site) the smallest
-            vertical displacment neighbour will be returned.
+            vertical_bias : string/None
```
Comment (carolinejjones) [inline lib/improver/spotdata/neighbour_finding.py]: vertical_bias (str or None):

File: lib/improver/spotdata/neighbour_finding.py [hunk 8]
```diff
@@ -63,19 +63,18 @@ def __init__(self, method='fast_nearest_neighbour',
         the grid points closest to sites of interest.
 
         Args:
-        -----
-        method : string
-            Name of the method of neighbour finding to be used.
+            method : string
```
Comment (carolinejjones) [inline lib/improver/spotdata/neighbour_finding.py]: method (string):

File: lib/improver/spotdata/neighbour_finding.py [hunk 9]
```diff
@@ -63,19 +63,18 @@ def __init__(self, method='fast_nearest_neighbour',
...
+            vertical_bias : string/None
+                Sets the preferred vertical displacement bias of the grid point
+                relative to the site; above/below/None. If this criteria cannot
+                be met (e.g. bias below, but all grid points above site) the
+                smallest vertical displacment neighbour will be returned.
 
-        land_constraint : boolean
-            If True spot data sites on land should only select neighbouring
-            grid points also over land.
+            land_constraint : boolean
```
Comment (carolinejjones) [inline lib/improver/spotdata/neighbour_finding.py]: land_constraint (Boolean):

File: lib/improver/tests/utilities/test_cube_manipulation.py [hunk 10]
```diff
@@ -108,12 +108,12 @@ def set_up_percentile_temperature_cube():
...
        as scalar or auxiliary coordinate.
-       Parameters
-       ----------
-       cube: cube
-           Iris cube containing coordinates to be checked
-       coord: coordinate
-           Cube coordinate to check
+
+       Args:
+           cube: cube
```
Comment (carolinejjones) [inline lib/improver/tests/utilities/test_cube_manipulation.py]: cube (iris.cube.Cube):

File: lib/improver/tests/wind_downscaling/test_roughnesscorrection.py [hunk 11]
```diff
@@ -123,20 +123,19 @@ def __init__(self, nx_ny=3, AoS=None, Sigma=None, z_0=0.2, pporog=None,
...
-        Sigma: float or 1D or 2D array
-            Standard deviation field of height in grid cell
-        z_0: float or 1D or 2D array
-            Vegetative roughness field
-        pporog: float or 1D or 2D array
-            Unsmoothed orography field on post-processing grid
-        modelorog: float or 1D or 2D array
-            Model orography field on post-processing grid
+        Args:
+            nx_ny: a scalar or an np.array([x,y])
```
Comment (carolinejjones) [inline lib/improver/tests/wind_downscaling/test_roughnesscorrection.py]: incorrect format

File: lib/improver/convection.py [hunk 12]
```diff
@@ -54,42 +54,42 @@ def __init__(
...
-            ens_factor : float
+            ens_factor (float):
                 The factor with which to adjust the neighbourhood size
                 for more than one ensemble member.
                 If ens_factor = 1.0 this essentially conserves ensemble
                 members if every grid square is considered to be the
                 equivalent of an ensemble member.
                 Optional, defaults to 1.0
-            use_adjacent_grid_square_differences : Logical
+            use_adjacent_grid_square_differences (Logical):
```
[the remark this answers is not in the document]
Comment (carolinejjones) [reply lib/improver/convection.py]: Ignore this type of comments. I've just noticed in your description above that you are not trying to fix these sorts of errors in this pull request.

**Comment by wxtim** [comment]
I cannot identify anything that @carolinejjones  hasn't seen.

**Comment by carolinejjones** [comment]
Ran tests and passed successfully. Creating the documentation doesn't cause any warnings to be raised.

**Pull request merged**